src/pages/menuUpload/orderDetails/OrderLineDetails.tsx

```diff
@@ -0,0 +1,280 @@
+            product: newItemName.id,
+            item_details: lineItem.item_details,
+            quantity: lineItem.quantity,
+            old_quantity: lineItem.old_quantity,
+            product_name: newItemName.name,
+            batch_number: '',
+          } : item)),
+      ]);
+      setItemName(newItemName);
+    } else if (newItemName === null) {
+      return;
+    }
+
+    setItemName(newItemName);
+  };
+
+  const handleNewItemBatchNumber = (newItemBatchNumber: any, lineItem: any) => {
+    if (newItemBatchNumber !== null && lineItem !== null && lineItem !== undefined) {
+      setLineItems([
+        ...lineItems.map((item) => (item.id === lineItem.id
+          ? {
+            id: lineItem.id,
+            order_details: lineItem.order_details,
+            product: lineItem.product,
+            item_details: newItemBatchNumber.id,
+            quantity: lineItem.quantity,
+            old_quantity: lineItem.old_quantity,
+            product_name: lineItem.product_name,
+            batch_number: newItemBatchNumber.batch_number,
+          } : item)),
+      ]);
+    }
+  };
+
+  const removeLineItem = (lineItemId: number) => {
+    setLineItems([
+      ...lineItems.filter(({ id }) => id !== lineItemId),
+    ]);
+  };
+
+  return (
+    <EditableTable
+      buttonTitle={format('app.add_item')}
+      onAddButtonClick={addNewItem}
+      columnHeaders={columnHeaders}
+    >
+      {lineItems?.map((lineItem: any) => (
+        <div className="flex h-12 px-8 items-center even:bg-transit-grey-light" key={lineItem.id}>
+          <div className="w-1/3 pr-2">
+            <ItemPickerOutsideForm
+              field={mode === 'Add' ? { id: Math.floor(Math.random() * 10000), name: null, lineItem } : { id: lineItem.product, name: lineItem.product_name, lineItem }}
+              onChangeItemName={handleNewItemName}
+            />
+          </div>
+          <div className="w-1/3 pr-2">
+            <BatchNumberPicker
+              field={mode === 'Add' ? {
+                id: lineItem.id,
+                batch_number: lineItem.batch_number,
+                lineItem,
+              } : {
+                id: lineItem.item_details,
+                batch_number: lineItem.batch_number,
+                lineItem,
+              }}
+              watch={itemName}
+              onChangeBatchNumber={handleNewItemBatchNumber}
+            />
+          </div>
+          <div className="w-1/3 pr-2">
+            <Input
+              type="number"
+              className="w-full"
+              isInvalid={lineItem.quantity.trim().length < 1}
+              placeholder={format('order_details.total_quantity.label')}
+              defaultValue={mode === 'Add' ? null : lineItem.quantity}
+              onChange={(event: any) => {
+                handleUpdateQuantity(event.target.value, lineItem);
+              }}
+            />
+          </div>
+          <div>
+            {/* @ts-ignore */}
```

Comment:
Why here there is 'ts-ignore'? Are there any problem with type for this 'component'?

Reply:
Nope, I forget to remove it :)